lib/fastlane/plugin/wpmreleasetoolkit/actions/common/buildkite_trigger_build_action.rb

```diff
@@ -0,0 +1,90 @@
+module Fastlane
+  module Actions
+    class BuildkiteTriggerBuildAction < Action
+      def self.run(params)
+        require 'buildkit'
+
+        UI.message "Triggering build on branch #{params[:branch]} at #{params[:commit]}"
+
+        pipeline_name = {
+          PIPELINE: params[:pipeline_file]
+        }
+
+        client = Buildkit.new(token: params[:buildkite_token])
+        response = client.create_build(
+          params[:buildkite_organization],
+          params[:buildkite_pipeline],
+          {
+            branch: params[:branch],
+            commit: params[:commit],
+            env: params[:environment].merge(pipeline_name)
+          }
+        )
+
+        response.state == 'scheduled' ? UI.message('Done!') : UI.user_error!("Failed to start job\nError: [#{response}]")
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        'Triggers a job on Buildkite'
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(
+            key: :buildkite_token,
+            env_name: 'BUILDKITE_ACCESS_TOKEN',
+            description: 'Buildkite Personal Access Token',
+            type: String
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :buildkite_organization,
+            env_name: 'BUILDKITE_ORGANIZTION',
+            description: 'The Buildkite organization that contains your pipeline',
+            type: String,
+            default_value: 'wordpress-mobile'
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :buildkite_pipeline,
+            env_name: 'BUILDKITE_PIPELINE',
+            description: 'The Buildkite pipeline you\'d like to build',
+            type: String
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :branch,
+            description: 'The branch you want to build',
+            type: String
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :commit,
+            description: 'The commit hash you want to build',
+            type: String,
+            default_value: 'HEAD'
+          ),
```

Comment:
Should we [make those exclusive to one another](https://github.com/fastlane/fastlane/blob/master/fastlane_core/lib/fastlane_core/configuration/config_item.rb#L48-L49), so that only one is valid? (Otherwise, what happens if we provide both?!)

```suggestion
          FastlaneCore::ConfigItem.new(
            key: :branch,
            description: 'The branch you want to build',
            type: String,
            conflicting_options: [:commit]
          ),
          FastlaneCore::ConfigItem.new(
            key: :commit,
            description: 'The commit hash you want to build',
            type: String,
            default_value: 'HEAD'
            conflicting_options: [:branch]
          ),
```

Reply:
[Both are required by the Buildkite API](https://buildkite.com/docs/apis/rest-api/builds#create-a-build), so they can't be mutually exclusive.

Reply:
Oh. How strange… I wonder how Buildkite reacts then if you provide a commit sha1 that is not on the branch you also provide but on a different branch instead… 🤔

Reply:
Looks like Buildkite prioritizes the commit and... just trust you know the branch exists?

![image](https://user-images.githubusercontent.com/1218433/152204848-93bbc2ae-17cb-453b-ad6e-66fc0231f326.png)

I started it with:

```
➜ bf run buildkite_trigger_build buildkite_organization:automattic \
  buildkite_pipeline:wordpress-ios \
  branch:will-this-be-ignored-question-mark \
  commit:b461d3c3eff4aba1cf91ca0482e7c05bfc9454e1 \
  pipeline_file:pipeline.yml
```

I should have used a commit different from the tip of the branch it belongs to, but the test turned out interesting anyway 🤔 

I wonder if that's a bug in the API?

Reply:
Well the docs Jeremy linked to says:
> [Branch]: Branch the commit belongs to. This allows you to take advantage of your pipeline and step-level branch filtering rules.

And given a commit can belong to multiple branches (example case being a train of PRs, with branch B built on top of branch A, so commit at tip of branch A would be in both A and B), `git branch -a --contains <sha1>` does not always return a single result, so I guess that makes sense for Buildkite to request us to tell which branch this is supposed to be triggered from in order to apply the correct branch filtering rules accordingly 🤷

Reply:
This is probably the correct answer, because it seems weird otherwise!